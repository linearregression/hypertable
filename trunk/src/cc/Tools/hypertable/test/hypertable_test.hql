CREATE TABLE hypertable (
apple,
banana
);
insert into hypertable VALUES ('2007-12-02 08:00:00', 'foo', 'apple:0', 'nothing'), ('2007-12-02 08:00:01', 'foo', 'apple:1', 'nothing'), ('2007-12-02 08:00:02', 'foo', 'apple:2', 'nothing');
insert into hypertable VALUES ('2007-12-02 08:00:03', 'foo', 'banana:0', 'nothing'), ('2007-12-02 08:00:04', 'foo', 'banana:1', 'nothing'), ('2007-12-02 08:00:05', 'bar', 'banana:2', 'nothing');
select * from hypertable;
delete "apple:1" from hypertable where row = 'foo' timestamp '2007-12-02 08:00:01';
select * from hypertable;
delete banana from hypertable where row = 'foo';
select * from hypertable;
insert into hypertable VALUES ('how', 'apple:0', 'nothing'), ('how', 'apple:1', 'nothing'), ('how', 'apple:2', 'nothing');
insert into hypertable VALUES ('now', 'banana:0', 'nothing'), ('now', 'banana:1', 'nothing'), ('now', 'banana:2', 'nothing');
quit
