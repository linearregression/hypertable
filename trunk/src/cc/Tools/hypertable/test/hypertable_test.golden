CREATE TABLE hypertable (
apple,
banana
);
insert into hypertable VALUES ('2007-12-02 08:00:00', 'foo', 'apple:0', 'nothing'), ('2007-12-02 08:00:01', 'foo', 'apple:1', 'nothing'), ('2007-12-02 08:00:02', 'foo', 'apple:2', 'nothing');
insert into hypertable VALUES ('2007-12-02 08:00:03', 'foo', 'banana:0', 'nothing'), ('2007-12-02 08:00:04', 'foo', 'banana:1', 'nothing'), ('2007-12-02 08:00:05', 'bar', 'banana:2', 'nothing');
select * from hypertable;
2007-12-02 08:00:05.000000000	bar	banana:2	nothing
2007-12-02 08:00:03.000000000	foo	banana:0	nothing
2007-12-02 08:00:04.000000000	foo	banana:1	nothing
2007-12-02 08:00:00.000000000	foo	apple:0	nothing
2007-12-02 08:00:01.000000000	foo	apple:1	nothing
2007-12-02 08:00:02.000000000	foo	apple:2	nothing
delete "apple:1" from hypertable where row = 'foo' timestamp '2007-12-02 08:00:01';
delete: row='foo' family=2 qualifier='1' ts=1196611201000000001
select * from hypertable;
2007-12-02 08:00:05.000000000	bar	banana:2	nothing
2007-12-02 08:00:03.000000000	foo	banana:0	nothing
2007-12-02 08:00:04.000000000	foo	banana:1	nothing
2007-12-02 08:00:00.000000000	foo	apple:0	nothing
2007-12-02 08:00:02.000000000	foo	apple:2	nothing
delete banana from hypertable where row = 'foo';
delete: row='foo' family=1 ts=0
select * from hypertable;
2007-12-02 08:00:05.000000000	bar	banana:2	nothing
2007-12-02 08:00:00.000000000	foo	apple:0	nothing
2007-12-02 08:00:02.000000000	foo	apple:2	nothing
insert into hypertable VALUES ('how', 'apple:0', 'nothing'), ('how', 'apple:1', 'nothing'), ('how', 'apple:2', 'nothing');
insert into hypertable VALUES ('now', 'banana:0', 'nothing'), ('now', 'banana:1', 'nothing'), ('now', 'banana:2', 'nothing');
